src/channel_internal.c

```diff
@@ -279,148 +353,166 @@ static int dequeue_operation(THANDLE(CHANNEL_INTERNAL) channel_internal, THANDLE
     PDLIST_ENTRY op_entry = DList_RemoveHeadList(&channel_internal_ptr->op_list);
 
     CHANNEL_OP* channel_op = CONTAINING_RECORD(op_entry, CHANNEL_OP, anchor);
-    if (pull_callback != NULL)
+    if (on_data_available_cb != NULL)
     {
-        /*Codes_SRS_CHANNEL_INTERNAL_43_112: [ channel_internal_pull shall store the correlation_id, pull_callback and pull_context in the obtained operation. ]*/
+        /*Codes_SRS_CHANNEL_INTERNAL_43_112: [ channel_internal_pull shall store the correlation_id, on_data_available_cb and pull_context in the obtained operation. ]*/
         THANDLE_INITIALIZE(RC_STRING)(&channel_op->pull_correlation_id, pull_correlation_id);
-        channel_op->pull_callback = pull_callback;
+        channel_op->on_data_available_cb = on_data_available_cb;
         channel_op->pull_context = pull_context;
```

Comment:
It looks like you're replacing the old callback with a new callback.  What happens with the old context?  Does that leak?

Reply:
The callback is being set for the first time. It was NULL before this.